src/test/java/co/rsk/federate/signing/hsm/client/PowHSMSigningClientBtcTest.java

```diff
@@ -16,80 +21,344 @@
         client = new PowHSMSigningClientBtc(hsmClientProtocol, HSM_VERSION);
     }
 
     @Test
     void signOk() throws Exception {
-        ObjectNode expectedPublicKeyRequest = buildGetPublicKeyRequest();
+        ObjectNode expectedPublicKeyRequest = buildGetPublicKeyCommand(HSM_VERSION);
         ObjectNode publicKeyResponse = buildResponse(0);
-        publicKeyResponse.put(PUB_KEY.getFieldName(), "001122334455");
+        publicKeyResponse.put(PUB_KEY.getFieldName(), publicKey);
         when(jsonRpcClientMock.send(expectedPublicKeyRequest)).thenReturn(publicKeyResponse);
 
         PowHSMSignerMessage messageForSignature = buildMessageForIndexTesting(0);
 
-        ObjectNode expectedSignRequest = buildSignRequest(messageForSignature);
-        ObjectNode response = buildSignResponse("223344", "55667788", 0);
+        ObjectNode expectedSignRequest = buildSignCommand(HSM_VERSION, messageForSignature);
+        ObjectNode response = buildSignResponse(0);
 
         when(jsonRpcClientMock.send(expectedSignRequest)).thenReturn(response);
 
-        HSMSignature signature = client.sign("a-key-id", messageForSignature);
+        HSMSignature signature = client.sign(keyId, messageForSignature);
 
-        assertArrayEquals(Hex.decode("223344"), signature.getR());
-        assertArrayEquals(Hex.decode("55667788"), signature.getS());
-        assertArrayEquals(Hex.decode("001122334455"), signature.getPublicKey());
+        assertArrayEquals(Hex.decode(R_SIGNATURE_VALUE), signature.getR());
+        assertArrayEquals(Hex.decode(S_SIGNATURE_VALUE), signature.getS());
+        assertArrayEquals(Hex.decode(publicKey), signature.getPublicKey());
         verify(jsonRpcClientMock, times(1)).send(expectedSignRequest);
         verify(jsonRpcClientMock, times(1)).send(expectedPublicKeyRequest);
     }
 
+    private BtcTransaction createPegout(List<Coin> outpointValues, Address destinationAddress,
+        boolean segwit) {
+        BtcTransaction fundingTransaction = new BtcTransaction(btcMainnetParams);
+        fundingTransaction.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX,
+            new Script(new byte[]{}));
+
+        for (Coin outpointValue : outpointValues) {
+            fundingTransaction.addOutput(outpointValue, oldFederation.getAddress());
+        }
+
+        BtcTransaction pegoutBtcTx = new BtcTransaction(btcMainnetParams);
+
+        Script inputScriptThatSpendsFromTheFederation = createBaseInputScriptThatSpendsFromTheFederation(
+            oldFederation);
+
+        Coin fee = Coin.MILLICOIN;
+        for (int inputIndex = 0; inputIndex < outpointValues.size(); inputIndex++) {
+            TransactionInput addedInput = pegoutBtcTx.addInput(
+                fundingTransaction.getOutput(inputIndex));
+            addedInput.setScriptSig(inputScriptThatSpendsFromTheFederation);
+            if (segwit) {
+                addWitness(pegoutBtcTx, inputIndex);
```

Comment:
Well, if I don't set the `scriptSig` tests fail because of [this](https://github.com/rsksmart/powpeg-node/blob/68443626a507790ea1b32f642a99fe7f5ebff84d/src/main/java/co/rsk/federate/signing/hsm/message/SignerMessageBuilder.java#L24). I think we will have to stick to this logic until we fully implement segwit support in the powpeg and bitcoinj-thin. Wdyt?